<div style="background:linear-gradient(135deg,#042f2e 0%,#0f766e 55%,#14b8a6 100%);border-radius:18px;padding:34px 30px;color:#fff;font-family:Inter,Segoe UI,sans-serif">
  <div style="font-size:12px;letter-spacing:3px;color:#99f6e4;font-weight:700;text-transform:uppercase">Chapter 41 · Solutions</div>
  <div style="font-size:34px;font-weight:900;line-height:1.1;margin:10px 0 6px">Distributions for Inference, Worked Answers ✅</div>
  <div style="font-size:15px;color:#d1fae5;max-width:720px;line-height:1.6">Full solutions to the five challenges. Try them yourself first, then compare.</div>
  <div style="margin-top:16px;font-size:13px;color:#a7f3d0">Statistics, Data Science and AI: A Visual Handbook · John Fisher · 2026</div>
</div>

## ⚙️ Setup

In [1]:
import numpy as np
from scipy import stats
rng = np.random.default_rng(411)
print("ready")

ready


<div style="background:#ecfdf5;border-left:5px solid #0d9488;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#0d9488;letter-spacing:1px">CHALLENGE 1 · t CRITICAL VALUE</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">Why small samples need more</div>
<div style="color:#4a5578;margin-top:6px">For a 95% two-sided test with a sample of n = 10 (df = 9), find the critical t value and compare it to the normal's 1.96.</div>
</div>

In [2]:
df = 9
tcrit = stats.t.ppf(0.975, df)
print(f"t critical (df=9) = {tcrit:.3f}")
print(f"normal z          = 1.960")

t critical (df=9) = 2.262
normal z          = 1.960


**Answer:** the critical value is t = **2.262**, noticeably larger than z = 1.96. With only 9 degrees of freedom the t has heavier tails, so a small sample must clear a higher bar to reach significance, the price of estimating the spread from little data.

<div style="background:#ecfdf5;border-left:5px solid #0d9488;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#0891b2;letter-spacing:1px">CHALLENGE 2 · CHI-SQUARE MEAN</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">Sum of squared normals</div>
<div style="color:#4a5578;margin-top:6px">A chi-square distribution is the sum of df squared standard normals. Confirm by simulation that chi-square(5) has mean 5.</div>
</div>

In [3]:
built = (rng.normal(0,1,size=(300_000,5))**2).sum(axis=1)
print(f"simulated mean of chi-square(5) = {built.mean():.3f}  (theory = 5)")

simulated mean of chi-square(5) = 4.996  (theory = 5)


**Answer:** the simulated mean is about **5.0**, matching the rule that a chi-square's mean equals its degrees of freedom. Each squared standard normal contributes a mean of 1, and five of them sum to 5.

<div style="background:#ecfdf5;border-left:5px solid #0d9488;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#0d9488;letter-spacing:1px">CHALLENGE 3 · t CONFIDENCE INTERVAL</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">Small-sample interval for a mean</div>
<div style="color:#4a5578;margin-top:6px">A sample of n = 10 has mean 50 and sample sd 8. Build a 95% confidence interval for the mean using the t distribution.</div>
</div>

In [4]:
xbar, s, n = 50, 8, 10
se = s/np.sqrt(n)
tcrit = stats.t.ppf(0.975, n-1)
lo, hi = xbar - tcrit*se, xbar + tcrit*se
print(f"SE = 8/sqrt(10) = {se:.3f}, t(df=9) = {tcrit:.3f}")
print(f"95% CI = 50 +/- {tcrit:.3f}*{se:.3f} = [{lo:.2f}, {hi:.2f}]")

SE = 8/sqrt(10) = 2.530, t(df=9) = 2.262
95% CI = 50 +/- 2.262*2.530 = [44.28, 55.72]


**Answer:** SE = 8/&#8730;10 &approx; 2.53, and with t(9) = 2.262 the interval is 50 &plusmn; 2.262(2.53) = **[44.28, 55.72]**. Using t instead of z makes the interval a bit wider, the honest cost of not knowing the true standard deviation.

<div style="background:#ecfdf5;border-left:5px solid #0d9488;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#db2777;letter-spacing:1px">CHALLENGE 4 · GOODNESS OF FIT</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">A chi-square test</div>
<div style="color:#4a5578;margin-top:6px">A bag of candy should be 25% of each of 4 colors. In 200 candies you observe [40, 55, 52, 53]. Test whether the colors are equally likely.</div>
</div>

In [5]:
observed = np.array([40, 55, 52, 53])
expected = np.full(4, 200/4)
chi2_stat, p = stats.chisquare(observed, expected)
print(f"chi-square = {chi2_stat:.3f}, df = 3, p-value = {p:.3f}")
print("-> consistent with equal colors" if p>0.05 else "-> evidence colors are NOT equal")

chi-square = 2.760, df = 3, p-value = 0.430
-> consistent with equal colors


**Answer:** the chi-square statistic is **2.76** with df = 3, giving p &approx; **0.430**. Since p &gt; 0.05, the data is consistent with equal colors; the deviations from 50 each are within normal sampling variation.

<div style="background:#ecfdf5;border-left:5px solid #0d9488;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#0f766e;letter-spacing:1px">CHALLENGE 5 · COMPARE TWO MODELS</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">A paired t-test</div>
<div style="color:#4a5578;margin-top:6px">Two models are scored on 8 cross-validation folds. Use a paired t-test to decide whether model B is significantly better than model A.</div>
</div>

In [6]:
A = np.array([0.72,0.75,0.71,0.74,0.70,0.73,0.72,0.74])
B = np.array([0.75,0.78,0.74,0.77,0.74,0.76,0.75,0.78])
t_stat, p = stats.ttest_rel(B, A)
print(f"mean A = {A.mean():.3f}, mean B = {B.mean():.3f}")
print(f"paired t = {t_stat:.3f}, p-value = {p:.5f}")
print("-> B is significantly better" if p<0.05 else "-> not significant")

mean A = 0.726, mean B = 0.759
paired t = 19.858, p-value = 0.00000
-> B is significantly better


**Answer:** model B beats A on every fold, and the paired t-test gives a p-value far below 0.05, so the improvement is **statistically significant**. Pairing by fold removes the fold-to-fold difficulty and isolates the model effect, the standard way to compare algorithms fairly.

---
<div style="background:#ffffff;border:1px solid #e6e9f2;border-radius:16px;padding:22px 26px;font-family:Inter,sans-serif">
<div style="font-size:19px;font-weight:800;color:#1a2138">🎉 Nicely done</div>
<div style="color:#4a5578;line-height:1.8;margin-top:8px">You found a t critical value, confirmed the chi-square mean, built a small-sample t interval, ran a goodness-of-fit test, and compared two models with a paired t-test. The t, chi-square, and F are the everyday tools of inference, and you will use them throughout the chapters ahead.</div>
</div>

<div style="text-align:center;color:#8b94b3;font-size:12px;margin-top:16px">Statistics, Data Science and AI: A Visual Handbook · © 2026 John Fisher</div>